File: tutorial_src/gue/public/gue_lib.cpp
#include <iostream>

#include "gue.h"

using namespace std;

int N, X;

int init() {
    cout << "Input n: ";
    cin >> N;
    cout << "Input secret number: ";
    cin >> X;
    
    return N;
}

bool isGreater(int x) {
    cout << "Query: " << x << "\n";
    return X > x;
}

void answer(int x) {
    if (x == X) {
        cout << "Program guessed correctly!\n";
    } else {
        cout << "Program answered " << x << ", while the correct answer is " << X << "\n";
    }
}

File: tutorial_src/gue/public/guelib.py
N = 0
X = 0

def init():
    global N
    global X
    print("Input n: ")
    N = int(input())
    print("Input secret number: ")
    X = int(input())
    
    return N

def isGreater(x):
    global X
    print("Query: " + str(x));
    return X > x;

def answer(x):
    global X
    if x == X:
        print("Program guessed correctly!")
    else:
        print("Program answered " + str(x) + ", while the correct answer is " + str(X));
